diff --git a/src/parse.js b/src/parse.js
index 966f17046167769be5a2d8ec674b084baddae5f9..3513e074a91c9201d8cf5b77d1175341e590e6e1 100644
--- a/src/parse.js
+++ b/src/parse.js
@@ -726,8 +726,12 @@ function parse(source, root, options) {
                     var lastValue;
                     if (skip("[", true)) {
                         do {
-                            lastValue = readValue(true);
-                            value.push(lastValue);
+                            if (peek() === "{")
+                                parseOptionValue({}, name + "." + token); // consume without recording
+                            else {
+                                lastValue = readValue(true);
+                                value.push(lastValue);
+                            }
                         } while (skip(",", true));
                         skip("]");
                         if (typeof lastValue !== "undefined") {
